provider/parameter.go

```diff
@@ -429,84 +450,201 @@ func (v *Parameter) Valid() diag.Diagnostics {
 		}
 	}
 
-	optionNames := map[string]any{}
-	optionValues := map[string]any{}
-	if len(v.Option) > 0 {
-		for _, option := range v.Option {
-			_, exists := optionNames[option.Name]
-			if exists {
-				return diag.Diagnostics{{
+	optionValues, diags := v.ValidOptions(optionType, mode)
+	if diags.HasError() {
+		return "", diags
+	}
+
+	if mode == ValidationModeTemplateImport && v.Default != nil {
+		// Template import should validate the default value.
+		err := valueIsType(v.Type, *v.Default)
+		if err != nil {
+			return "", diag.Diagnostics{
+				{
+					Severity:      diag.Error,
+					Summary:       fmt.Sprintf("Default value is not of type %q", v.Type),
+					Detail:        err.Error(),
+					AttributePath: defaultValuePath,
+				},
+			}
+		}
+
+		d := v.validValue(*v.Default, optionType, optionValues, defaultValuePath)
+		if d.HasError() {
+			return "", d
+		}
+	}
+
+	// TODO: This is a bit of a hack. The current behavior states if validation
+	//   is given, then apply validation to unset values.
+	//   This should be removed, and all values should be validated. Meaning
+	//   value == nil should not be accepted in the first place.
+	if len(v.Validation) > 0 && value == nil {
+		empty := ""
+		value = &empty
+	}
```

Comment:
I feel like this would be more readable if you inverted the `if` clause and had it `return` in the case that we don't do validation.

Reply:
I can do that :+1: